### ISSUE TITLE
[sprint - rolemanager?] `.sprint follow` returns a "sorry, I'm still loading the ... module" error

### ISSUE THREAD

**Issue by Laogeodritt** [Worldbuilding/kaztron#227]

**Comment by Laogeodritt** [comment]
Suspect that this is due to 1) kaztron disconnecting and reconnecting, with 2) add_managed_role detecting an existing managed role on the reconnect startup, but 3) the existing managed role being assigned to a previous cog instance, which is no longer considered ready.

This would explain the discrepancy between the sprint module being ready and other commands usable, but the managed role command states that the same sprint cog is not ready yet.

However this isn't 100% consistent: given that RoleManager.managed_roles is an instance attribute, not a class attribute, this relies on other cogs being reinstanced but RoleManager reusing the same instance.

**Comment by Laogeodritt** [comment]
Solution for testing, add to RoleManager cog:

```python
    def unload_kazcog(self):
        """ Unload managed roles. """
        logger.debug("Unloading managed roles...")
        for role_name, mr in self.managed_roles.values():
            for cmd in mr.commands:  # type: commands.Command
                if cmd.parent is not None:
                    parent = cmd.parent  # type: commands.GroupMixin
                else:
                    parent = self.bot
                rmret = parent.remove_command(cmd.name)

                if rmret is not None:
                    logger.debug("Unloaded command {} for role {}".format(cmd.name, role_name))
                else:
                    logger.warning("Failed to unload command {} for role {}"
                        .format(cmd.name, role_name))
```

This should unload all the commands (not the dummy groups - possible improvement) from the role manager upon unload, which should ensure that they are not loaded during the startup process after a disconnect/reconnect. Need to do some longer-term testing to see if this changes the behaviour.

**Comment by Laogeodritt** [comment]
Not observed again on the test instance. Closed for 2.2 release; to be monitored in deployed instances.
